src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +137,22 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+  //get the info object from the server 
+  public ServerInfo getInfo(){
+    try(final Connection connection = this.source.connect()){
```

Comment:
You should change the name of the function to something like `getVersion()` to make it more descriptive for anyone reading the code